### PR TITLE
fix(core,react): preserve composer draft when suggestion send fires mid-run

### PR DESCRIPTION
**Pull request by ephraimduncan**
## What

`SuggestionPrimitive.Trigger` (and `ThreadPrimitive.Suggestion`) with `send` destroyed the user's composer draft when clicked while a run was in progress: the `resolvedSend && !isRunning` condition fell through to the `else` branch, which called `composer.setText(prompt)` (or appended the prompt with `clearComposer={false}`).

## Why

Clicking a "send" suggestion during a run should never mutate the composer — the composer-mutation branch is the `send={false}` behavior. The fallthrough silently replaced whatever the user had typed.

## How

- `send=true` while running now no-ops unless the thread has the `queue` capability — the exact gate the composer send path already uses (`ComposerInput`, `useComposerSend`: `isRunning && !capabilities.queue`). With queue support, `thread.append` proceeds and enqueues, matching composer behavior.
- Fixed in both copies of the logic: the canonical `useSuggestionTrigger` in `@assistant-ui/core` (used by `ThreadPrimitive.Suggestion`, react-native, react-ink) and the inline duplicate in `@assistant-ui/react`'s `SuggestionPrimitive.Trigger`.
- Added contract tests for `useSuggestionTrigger`; the draft-preservation and queue-enqueue tests fail against the pre-fix logic.

## Verification

- New tests 5/5; verified 2 of them fail with the old fallthrough restored.
- `tsc --noEmit` clean for touched files in core and react; oxlint/oxfmt clean.
- Core suite: 770/772 — the 2 failures are pre-existing in `AssistantCloudThreadHistoryAdapter.test.tsx` (cloud persistence, unrelated).

<a href="https://rupic.dev/s/s2.C53lN53R-Y9nL4slDob4xYGNG0fglNASz7uFmL1BHHJrWsDQ_gIrRf_PGyY?ref=github" target="_blank" rel="noopener noreferrer"><picture><source media="(prefers-color-scheme: dark)" srcset="https://rupic.dev/badges/track-dark-v1.svg">Track in Rupic</picture></a>

<a href="https://cubic.dev/pr/assistant-ui/assistant-ui/pull/5417?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by AVGVSTVS96** [comment]
@ephraimduncan thanks for this PR — the fix and the queue-aware handling were correct. I made some updates in 84ce432 before merging it:

- **Consolidated the fix into core.** `SuggestionPrimitive.Trigger` was carrying an inlined copy of the suggestion-trigger logic (a leftover from #3628, which migrated every other surface to the core hook). It now delegates to `useSuggestionTrigger` from `@assistant-ui/core/react` like `ThreadPrimitive.Suggestion`, react-native, and react-ink already do — so the fix lives in one place and the two copies can't drift again.
- **Disabled state instead of a silent no-op.** On runtimes without queue support, a `send` suggestion mid-run now renders disabled rather than swallowing the click, matching how `ComposerPrimitive.Send` handles the same state (`isRunning && !capabilities.queue`).
- **Test reworked to the house pattern.** Switched from mocking React's `useCallback` to `renderHook` (same shape as `useActionBarCopy.test.ts`), and added coverage for the `disabled` flag and a couple of edge cases.
- **Docs.** `primitives/suggestion.mdx` still described the old fallback behavior ("falls back to populating the composer") — updated it and the `clearComposer` JSDoc to describe the queue/disabled semantics.

File: packages/core/src/react/primitive-hooks/useSuggestionTrigger.ts [hunk 1]
```diff
@@ -17,9 +17,10 @@ export const useSuggestionTrigger = ({
   const resolvedSend = send ?? false;
 
   const trigger = useCallback(() => {
-    const isRunning = aui.thread.getState().isRunning;
+    if (resolvedSend) {
+      const { isRunning, capabilities } = aui.thread.getState();
+      if (isRunning && !capabilities.queue) return;
```
[the remark this answers is not in the document]
Comment (AVGVSTVS96) [reply packages/core/src/react/primitive-hooks/useSuggestionTrigger.ts:26]: Addressed by f18c167 (queued sends skip the clear) and pinned in 84ce432: the queue-path test now asserts `setText` is not called, so the draft-preservation contract is specified for both the idle and queued paths.

File: packages/react/src/primitives/suggestion/SuggestionTrigger.ts [hunk 2]
```diff
@@ -34,9 +34,10 @@ const useSuggestionTrigger = ({
   const resolvedSend = send ?? false;
 
   const callback = useCallback(() => {
```
[the remark this answers is not in the document]
Comment (AVGVSTVS96) [reply packages/react/src/primitives/suggestion/SuggestionTrigger.ts]: Done in 84ce432 — `SuggestionPrimitive.Trigger` now delegates to the core `useSuggestionTrigger` hook like the other surfaces, so the logic lives in one place.

File: packages/core/src/react/primitive-hooks/useSuggestionTrigger.ts [hunk 3]
```diff
@@ -17,14 +17,15 @@ export const useSuggestionTrigger = ({
...
```
[the remark this answers is not in the document]
Comment (AVGVSTVS96) [reply packages/core/src/react/primitive-hooks/useSuggestionTrigger.ts:26]: Took this in 84ce432: `disabled` now folds in `resolvedSend && isRunning && !capabilities.queue`, matching `useComposerSend`. Rode along with this fix since the silent no-op landed here too; changeset and docs mention the widened disabled state.

**Review by AVGVSTVS96 — APPROVED** [review]
